lib/instrumentation/transaction.js

```diff
@@ -112,17 +121,16 @@ Transaction.prototype.buildSpan = function () {
   }
 
   if (this.ended) {
-    this._agent.logger.debug('transaction already ended - cannot build new span %o', {id: this.id})
+    this._agent.logger.debug('transaction already ended - cannot build new span %o', {id: this.id}) // TODO: Should this be supported in the new API?
```

Comment:
When everything is fully streaming, I don't think there's any _reason_ to prevent further spans. It's only useful _without_ streaming to ensure the transaction doesn't stay open forever.

Reply:
The reason why I'm hesitant with this is that we might have a transaction that kicks off some form of background job that gets contextually attached to the transaction that started it. Any spans that's created as a result of this would get associated with the transaction. This will make the span waterfall almost useless as it can suddenly cover a period of days or weeks. Am I being too paranoid?

Reply:
That'd be user mode queuing though. We won't actually form that link unless we explicitly bind the callback. The logical continuation to the next job would be out-of-band from the request, it'd be triggered by the completion of another job in the queue.

I could see _maybe_ time-based jobs naively using `setTimeout` for everything, and _that_ maybe being an issue. But that's just not great design and we probably _should_ be drawing attention to it in some way. 🤔

Reply:
Yeah `setTimeout` and friends is probably the most common way to trigger this.

Question is however if the spans happening after the transaction ends are relevant to the user at all?

@elastic/apm-agent-devs @roncohen @makwarth In the new intake API, we get the ability to record spans that start after the parent transaction ends. Is this something we want to do or should those just be ignored? I.e. are they adding any value to the user or are they just noise?

Reply:
> I.e. are they adding any value to the user or are they just noise?

I think that depends on how the application is making use of that. I'd rather default to not ignore than to ignore these spans.

Reply:
If we don't ignore them by default, would it be a good idea to have a config option to ignore them?

It could even have 3 settings:

- Ignore spans started after the transaction ends
- Ignore spans started before the transaction ends, but ended after
- Allow all spans

Reply:
i understand you're worried about the case where people would set up a recurring periodic job from inside a transaction, because it would mean spans continue to happen in an endless stream. Two ideas:

* We could experiment in making the assumption that there are two categories of delayed execution `process.nextTick()/setImmediate()` and `setTimeout()/setInterval()`. The first one could indicate that the developer intended this to be part of the transaction, while the second category will start a new transaction.
* We could set a numerical limit to the number of spans that we will consider, after a transaction has ended - or a wall clock timeout which would be the limit to how much time we'll continue to "monitor" a transaction after is has ended.

Reply:
We could also include a special escape hatch that specifically calls to `setTimeout` or `setInterval` _after_ the transaction ends would _not_ continue the context. That way, anything _else_ in the code path will continue, but anything that could potentially be significantly delayed will _not_.